### PR TITLE
Fixes #1793: apoc.load.html procedure fail silently

### LINKED ISSUE #1793: apoc.load.html procedure fail silently
Currently, the `apoc.load.html` procedure returns an error and rolls back the transaction if any error occurs. The error messaging is not really helpful, you can look at https://github.com/neo4j-contrib/neo4j-apoc-procedures/blob/4.1/full/src/main/java/apoc/load/LoadHtml.java#L54.
In my opinion, it would be better to fail silently and just return an empty map with defined keys. Something along the line of:

```
  query.keySet().stream().forEach(key -> {
      output.put(key, empty map);
  });
```

### PR DESCRIPTION
**Pull request by vga91**
Fixes #1793

A brief list of proposed changes in order to fix the issue:

  - added `failsSilently: true/false` config (default `false`). With `true`, will be returned the list of the values, except for values not correctly parsed. In case of errors, will be printed a `log.warn`
  - improved the error returned, with `UnsupportedEncodingException`, `FileNotFoundException` and, in case of incorrect parse, with a message containing the incorrect tag

### CODE REVIEW

File: full/src/main/java/apoc/load/LoadHtml.java [hunk 1]
```diff
@@ -34,48 +37,61 @@
...
+
+                        result.put("children", getElements(element.children(), config));
+                    }
+                    else {
+                        if(element.hasText()) result.put("text", element.text());
+                    }
+
+                    elementList.add(result);
+            } catch (Exception e) {
+                final String parseError = "Error during parsing element: " + element;
```
Comment (fbiville) [inline full/src/main/java/apoc/load/LoadHtml.java:88]: maybe errors could be added to the output, in a reserved key? (maybe not for the scope of this PR/issue though)

File: full/src/test/java/apoc/load/LoadHtmlTest.java [hunk 2]
```diff
@@ -122,4 +125,65 @@ public void testQueryWithChildren() {
...
+            throw e;
+        }
+    }
+
+    @Test(expected = QueryExecutionException.class)
+    public void testQueryWithExceptionIfIncorrectUrl() {
+
+        final String filePath = new File("src/test/resources/wikipedia1.html").toURI().toString();
+        try {
+            String query = "CALL apoc.load.html('" + filePath + "',{a:'a'})";
```
Comment (fbiville) [inline full/src/test/java/apoc/load/LoadHtmlTest.java:165]: might be worth adding the variant with `failSilently: true` (just to "prove" it does not have an effect on unrecoverable errors)
same comment for the other test below

**Review by fbiville — APPROVED** [review]

File: full/src/main/java/apoc/load/LoadHtml.java (hunk 1 above)
Comment (vga91) [reply full/src/main/java/apoc/load/LoadHtml.java:88]: So, you mean, a structure like {`a: [listOfTagA]`, `h1: [list]`, .... `errors: [listOfErrorsIfExist]`} , right?
In this case maybe the `log.warn` at line 90 is useless

Comment (vga91) [reply full/src/main/java/apoc/load/LoadHtml.java:88]: Perhaps, we could leave both the log and the list and transform `failSilently` from boolean to enum, for example `FALSE, WITH_LOG, WITH_LIST`, and with `WITH_LOG`, we remains the `log.warn`, and with `WITH_LIST` we create the `errors: [listOfErrorsIfExist]`

Comment (fbiville) [reply full/src/main/java/apoc/load/LoadHtml.java:88]: sounds good!

Comment (vga91) [reply full/src/main/java/apoc/load/LoadHtml.java:98]: @fbiville Changed as previously described

**Pull request merged**